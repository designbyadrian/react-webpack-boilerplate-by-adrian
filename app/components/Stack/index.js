import React from 'react';
import PropTypes from 'prop-types';
import { inject, observer } from 'mobx-react';

import Card from 'components/Card';

@inject('postStore')
@observer
class Stack extends React.Component {
  static propTypes = {
    postStore: PropTypes.object,
  }

  componentDidMount() {
    this.props.postStore.loadPosts();
  }

  render() {
    const { posts, isLoading } = this.props.postStore;

    return (
      <div className="row">
        {posts.map(post => (
          <div key={post.id} className="col-12 col-sm-6 col-md-4 mb-1">
            <Card {...post} />
          </div>))}
      </div>
    );
  }
}

export default Stack;
