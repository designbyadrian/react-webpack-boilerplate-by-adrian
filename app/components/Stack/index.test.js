import React from 'react';
import { render } from 'react-testing-library';

import Stack from './index';

describe('components/Stack', () => {
  let store;

  beforeEach(() => {
    store = { postStore: { posts: [], loadPosts: jest.fn() } };
  });

  it('mounts', () => {
    const { container } = render(<Stack postStore={store.postStore} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
