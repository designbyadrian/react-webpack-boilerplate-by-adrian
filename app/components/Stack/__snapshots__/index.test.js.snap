// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Stack mounts 1`] = `
<div
  class="row"
/>
`;
